lib/actions/BackendAction.js

```diff
@@ -115,21 +116,28 @@ class BackendAction {
    * @returns {Promise<void>}
    */
   async writeExtensionConfigs () {
-    const attachedExtensions = await this.appSettings.loadAttachedExtensions()
+    const attachedExtensions = this.appSettings.loadAttachedExtensions()
     const extensionsFolder = path.join(this.appSettings.getApplicationFolder(), EXTENSIONS_FOLDER)
     const promises = []
-
     for (let extensionId in attachedExtensions) {
       const extensionPath = path.join(extensionsFolder, attachedExtensions[extensionId].path)
-      /** @type {ExtensionConfigJson} */
+      /** @type {Internal.ExtensionConfigJson} */
       const configJson = await fsEx.readJson(path.join(extensionPath, 'extension-config.json'))
-      /** @type {ExtensionConfig} */
+      /** @type {Internal.ExtensionConfig} */
       const config = { file: configJson, path: extensionPath }
-      promises.push(this._updateExtensionConfig(config, this.appSettings.getId()))
+      promises.push(this._updateExtensionConfig(config))
     }
     return Promise.all(promises)
   }
 
+  /**
+   * @param {Internal.ExtensionConfig} config
+   */
+  async _updateExtensionConfig (config) {
```

Comment:
(optional) We could move the function completely into line 128, since 128 became the only call of _updateExtensionConfig

Reply:
true, but this method is under test and it is still green, this would have to be refactored also